### PR TITLE
BUG: race condition in readContent at pkg/bundle/resources.go

### PR DESCRIPTION
**Pull request by nabam**
Shared global variable `getter.Getters` is updated in concurrently running goroutines. This causes corruption of authorization configuration for http getter when charts form various sources are used in the same git repository.

### CODE REVIEW

**Comment by nabam** [comment]
@StrongMonkey

File: pkg/bundle/resources.go [hunk 1]
```diff
@@ -306,13 +306,19 @@ func readContent(ctx context.Context, base, name string, auth Auth) (map[string]
...
+	}
+
 	c := getter.Client{
 		Ctx:     ctx,
 		Src:     name,
 		Dst:     temp,
 		Pwd:     base,
 		Mode:    getter.ClientModeDir,
-		Getters: getter.Getters,
+		Getters: getters,
```
Comment (mattfarina) [inline pkg/bundle/resources.go:321]: Further down in the code, the original `getter.Getters` is being updated. That should use the new `getters` instead. See...

https://github.com/rancher/fleet/blob/bd0fd515a2823c68b0ce0e5eab71558811fadd0c/pkg/bundle/resources.go#L350-L351

Can you please update all the uses of `getter.Getters`.

**Review by mattfarina — CHANGES_REQUESTED** [review]
Thanks for the pull request. I can see the collision. I found one place where there is still a problem in the code.

File: pkg/bundle/resources.go (hunk 1 above)
Comment (nabam) [reply pkg/bundle/resources.go:321]: this is `c.Getters` that already refer to copied getters map, but I'll change it anyway for the sake of readability

**Review by manno — APPROVED** [review]

**Pull request merged**